### DISCUSSION TITLE
the root (rather than sibling) /public folder is statically served when deploying a nextjs project to vercel

### DISCUSSION THREAD

**Discussion by shrugs** [vercel/next.js#23967] [Help]
# Bug report

The `public/` folder that is hosted statically when building a next.js project on vercel is always relative to the root of the repo, not the `public/` folder that is a sibling to the `next.config.js` used for building.

In development (`next dev`), the correct public directory is served.

## Describe the bug

My application consists of a collection of typescript packages and nextjs apps in a yarn workspaces monorepo. This monorepo contains an app (`gallery`) that is deployed to Vercel using the Next.JS preset, though it does not use the subpath as the root directory—because this is a yarn workspace, the correct way to install all of the relevant dependencies is `yarn install` in the root directory and setting the Vercel root directory as `packages/gallery` will run `yarn install` with a cwd of `packages/gallery` which will fail to install all of the relevant packages required to build the app (notably, it fails to install the dependencies for `packages/common` as well as makes it impossible to use lerna to build the projects according to their dependency tree (`lerna run build`)).

```
.
├── packages
│   ├── common
│   ├── dashboard
│   ├── db
│   ├── gallery
|   |   └── public
|   |       └── app.txt
│   ├── paint
│   ├── realtime
│   └── scripts
└── public
    └── root.txt
```

<img width="758" alt="image" src="https://user-images.githubusercontent.com/1535001/95878591-65fd4880-0d43-11eb-95a9-3e0c1ed3b156.png">
<img width="755" alt="image" src="https://user-images.githubusercontent.com/1535001/95880099-233c7000-0d45-11eb-8e7d-5481876c08cc.png">

## To Reproduce

Create a default yarn workspace monorepo and `create-next-app packages/app`. Create a `//public/root.txt` file at the root, and add a `packages/app/public/app.txt` file in the app's public folder.

To see that the correct public folder is used in development, run `lerna run dev` and access `http://localhost:3000/app.txt`. Note that `http://localhost:3000/root.txt` 404s.

Deploy the project to vercel, using settings similar to the above screenshot, making sure to specify the build directory as `packages/app/.next`. In the vercel build output, notice that `root.txt` is included and `app.txt` is not. Repeat the above GET requests on the vercel deployment to confirm that `/root.txt` is served and `app.txt` is not.

## Expected behavior

When building a nextjs project on vercel, the `/public` directory that is statically served should be a sibling directory of the `next.config.js` config file or, alternatively, a sibling directory of the `package.json` that defines the nextjs project.

## System information

- OS: macOS, Vercel
- Version of Next.js: v9.5.4
- Version of Node.js: v12.4.0

**Comment by AviKKi** [comment]
@shrugs  I'm facing same issue, were you able to fix it?

**Reply by shrugs** [reply]
i think i just implemented a copy-files step in my webpack build or something, sorry this was a year ago and i totally forgot 😓

**Reply by adophilus** [reply]
okay, I think I'll do same
